samples/videoToSequence/videotosequence.cpp

```diff
@@ -0,0 +1,504 @@
+    p_dec->ResetSaveFrameToFile();
+    decoding_complete = true;
+}
+
+
+void ShowHelpAndExit(const char *option = NULL) {
+    std::cout << "Options:" << std::endl
+    << "-i Input File / Folder Path - required" << std::endl
+    << "-o Output folder to dump sequences - dumps output if requested; optional" << std::endl
+    << "-d GPU device ID (0 for the first device, 1 for the second, etc.); optional; default: 0" << std::endl
+    << "-b seq_info.batch_size - specify the number of sequences to be decoded; (default: all sequences till eof)" << std::endl
+    << "-step - frame interval between each sequence; (default: sequence length)" << std::endl
+    << "-stride - distance between consective frames in a sequence; (default: 1)" << std::endl
+    << "-l - Number of frames in each sequence; (default: 3)" << std::endl
+    << "-crop crop rectangle for output (not used when using interopped decoded frame); optional; default: 0" << std::endl
+    << "-seek_mode option for seeking (0: no seek 1: seek to prev key frame); optional; default: 0" << std::endl
+    << "-m output_surface_memory_type - decoded surface memory; optional; default - 0"
+    << " [0 : OUT_SURFACE_MEM_DEV_INTERNAL/ 1 : OUT_SURFACE_MEM_DEV_COPIED/ 2 : OUT_SURFACE_MEM_HOST_COPIED/ 3 : OUT_SURFACE_MEM_NOT_MAPPED]" << std::endl;
+    exit(0);
+}
+// input_folder_path, output_folder_path, device_id, n_threads, seq_info, seek_mode, mem_type, argc, argv
+void ParseCommandLine(std::string &input_folder_path, std::string &output_folder_path, int &device_id, int &n_thread, SeqInfo &seq_info, int &seek_mode, 
+                bool &b_dump_output_frames, OutputSurfaceMemoryType &mem_type, int argc, char *argv[]) {
+    // Parse command-line arguments
+    if(argc <= 1) {
+        ShowHelpAndExit();
+    }
+    for (int i = 1; i < argc; i++) {
+        if (!strcmp(argv[i], "-h")) {
+            ShowHelpAndExit();
+        }
+        if (!strcmp(argv[i], "-i")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-i");
+            }
+            input_folder_path = argv[i];
+            continue;
+        }
+        if (!strcmp(argv[i], "-t")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-t");
+            }
+            n_thread = atoi(argv[i]);
+            if (n_thread <= 0 || n_thread > 64) {
+                ShowHelpAndExit(argv[i]);
+            }
+            continue;
+        }
+        if (!strcmp(argv[i], "-d")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-d");
+            }
+            device_id = atoi(argv[i]);
+            if (device_id < 0) {
+                ShowHelpAndExit(argv[i]);
+            }
+            continue;
+        }
+        if (!strcmp(argv[i], "-o")) {
+            if (++i == argc) {
+                ShowHelpAndExit("-o");
+            }
+            output_folder_path = argv[i];
+            if (!output_folder_path.empty()) {
+#if __cplusplus >= 201703L && __has_include(<filesystem>)
```

Comment:
@rrawther this is already added at he beginning of the file. We should not redefine. Please check